### PR TITLE
feat: trace_width modified

### PR DESCRIPTION
**Pull request by imrishabh18**

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -382,7 +382,6 @@ export const traceHintProps = z.object({
   order: z.number().optional(),
   offset: route_hint_point.optional(),
   offsets: z.array(route_hint_point).optional(),
-  traceWidth: z.number().optional(),
```
Comment (seveibar) [inline lib/index.ts:385]: hmm intentional? I think this should also be supported, you can do either, both options have different precedence

Comment (imrishabh18) [reply lib/index.ts:385]: Yeah, I am following the syntax where the traceWidth is added inside of the route. And then the default takes precedence

Comment (seveibar) [reply lib/index.ts:385]: so then `traceWidth` should exist here too right, both options should be supported.

Comment (seveibar) [reply lib/index.ts:385]: also `traceWidth` should use the `length` zod type, that way it can be defined like "0.1mm"

Comment (imrishabh18) [reply lib/index.ts:385]: Okay, we can add the support for this as well. Will make the update for the soup as well

Comment (seveibar) [reply lib/index.ts:385]: @imrishabh18 FWIW it's okay for the soup to pre-compute the width of each point on the route. The way to think about this is "props" is ergonomic for users and "soup" is ergonomic for _renderers_, we don't want the renderer to have to contain the logic to figure out what the width is at each point.

**Review by seveibar — APPROVED** [review]

**Pull request merged**